classes/class-dintero-checkout-gateway.php

```diff
@@ -52,8 +52,9 @@ public function __construct() {
 			 */
 			add_action(
 				'woocommerce_checkout_create_order_line_item',
-				function( $item, $cart_item_key ) {
+				function ( $item, $cart_item_key ) {
 					$item->update_meta_data( '_dintero_checkout_line_id', $cart_item_key );
+					$item->save();
 				},
 				10,
 				2
```

Comment:
Anonymous functions for actions should be avoided unless there is a good reason for it.